conslogging/conslogging.go

```diff
@@ -266,6 +272,67 @@ func (cl ConsoleLogger) PrefixColor() *color.Color {
 	return cl.color(c)
 }
 
+// Prints a GitHub Actions summary message to GITHUB_STEP_SUMMARY
+func (cl *ConsoleLogger) PrintGHASummary(message string) {
+	if !cl.isGitHubActions {
+		return
+	}
+
+	path := os.Getenv("GITHUB_STEP_SUMMARY")
+	if path == "" {
+		return
+	}
+	file, err := os.OpenFile(path, os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0644)
+	if err != nil {
+		return
+	}
+	defer file.Close()
+	_, _ = file.WriteString(message + "\n")
+}
+
+// PrintGHAError constructs a GitHub Actions error message.
+// The `file`, `line`, and `col` parameters are optional.
+func (cl *ConsoleLogger) PrintGHAError(message string, details ...string) {
+	cl.mu.Lock()
+	defer cl.mu.Unlock()
+
+	file := ""
+	line := ""
+	col := ""
+
+	if len(details) >= 3 {
+		file, line, col = details[0], details[1], details[2]
+	}
+
+	if file != "" && line != "" && col != "" {
+		cl.printGithubActionsControl(Error, "file=%s,line=%s,col=%s,title=Error::%s", file, line, col, message)
+	} else {
+		cl.printGithubActionsControl(Error, "title=Error::%s", message)
+	}
+}
+
+type GHHeader string
+
+const (
+	Error    GHHeader = "::error"
+	Group    GHHeader = "::group::"
+	EndGroup GHHeader = "::endgroup::"
+)
+
+// Print GHA control messages like ::group and ::error
+func (cl ConsoleLogger) printGithubActionsControl(header GHHeader, format string, a ...any) {
+	if !cl.isGitHubActions {
+		return
+	}
+	// Assumes mu locked.
+	w := new(bytes.Buffer)
+	defer func() {
+		_, _ = w.WriteTo(cl.errW)
+	}()
+
+	fmt.Fprintf(w, string(header)+" "+format+"\n", a...)
```

Comment:
might be an overkill, but wondering if some callers might already add a newline?
In which case maybe we can conditionally add the newline here (`strings.HasSuffix(s, \"n")`)

Reply:
done